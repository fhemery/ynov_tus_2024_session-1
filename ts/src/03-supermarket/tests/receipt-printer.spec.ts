import {ReceiptPrinter} from "../src/ReceiptPrinter";
import {Receipt} from "../src/model/Receipt";
import {Product} from "../src/model/Product";

describe('Receipt printer', () => {
    let receiptPrinter : ReceiptPrinter;

    beforeEach(() => {
        receiptPrinter = new ReceiptPrinter();
    })

    it('should return an empty receipt with 0 as total when no product', () => {
       const printedReceipt = receiptPrinter.printReceipt(new Receipt());

       expect(printedReceipt).toMatchSnapshot();
    });

    it('should return a single product in the receipt when there is one product', () => {
        const receipt = new Receipt();
        receipt.addProduct(new Product("Apples", 2), 3, 2.50, 7.50 );

        const printedReceipt = receiptPrinter.printReceipt(receipt);

        expect(printedReceipt).toMatchSnapshot();
    });
});
