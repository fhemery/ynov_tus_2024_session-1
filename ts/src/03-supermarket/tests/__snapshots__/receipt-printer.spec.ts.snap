// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receipt printer should return a single product in the receipt when there is one product 1`] = `
"Apples                              7.50
  2.50 * 3.000

Total:                              7.50"
`;

exports[`Receipt printer should return an empty receipt with 0 as total when no product 1`] = `
"
Total:                              0.00"
`;
